### PR TITLE
Update Playwright to use a custom test fixture with global mocks

### PR DESCRIPTION
**Pull request by zackkrida**
Creates a `test` fixture which overrides the base Playwright test fixture. This allows for easily setting global route mocks and other settings. Most crucially it removes the thousands of logs cluttering the output about the failed proxying of the analytics server.

This PR also makes our provider mocks global and adds a global mock for the analytics event endpoint, to prevent that from outputting hundreds of proxy-related error logs during the run of the test suite.

`frontend/test/playwright/utils/test-fixture.ts` is the main thing to review here. The rest is just implementations of the new fixture (switching the import from playwright to our custom test fixture).

~This will be cleaned up and undrafted once #2194 is merged.~

## Testing instructions

- Observe tests are still working and logs are no longer cluttered with analytics event warnings
- Ensure provider APIs are still mocked correctly in all relevant tests
- Try to `import { test } from "@playwright/test"` and you should be able to observe a lint error.

### CODE REVIEW

File: documentation/frontend/reference/playwright_tests.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ just frontend/run test:playwright visual-regression -u
 The above will run only test files with `visual-regression` in the path and will
 update any snapshot tests due to the `-u` flag.
 
-## Visual Regression tests
+## Writing Visual Regression tests
 
 When writing visual regression tests, it is good practice to write tests for
```
Comment (zackkrida) [inline documentation/frontend/reference/playwright_tests.md:72]: TODO: Add something about the new fixture here.

**Review by obulat — APPROVED** [review]
Nice cleanup!

Is there a way to add an eslint rule to stop anyone from importing `test` from `@playwright/test`?

**Comment by zackkrida** [comment]
@obulat indeed, there is! I had already written it and forgot to push 🤦

File: .eslintrc.js [hunk 2]
```diff
@@ -208,6 +208,12 @@ module.exports = {
             message:
               "Do not use @vue/test-utils' `shallowMount`. Use `~~/test/unit/test-utils/render` instead which includes helpful context setup or @testing-library/vue's `render` directly.",
           },
+          {
+            selector:
+              "ImportDeclaration[source.value='@playwright/test']:has(ImportSpecifier[local.name='test'])",
+            message:
+              "Do not import 'test' from '@playwright/test'. Use 'import { test } from '~~/test/playwright/utils/test-fixture' instead.",
```
Comment (zackkrida) [inline .eslintrc.js:215]:
```suggestion
              "Do not import 'test' from '@playwright/test'. Use 'import { test } from '~~/test/playwright/utils/test-fixture' instead, which mocks Openverse provider APIs and the analytics server.",
```

File: frontend/test/playwright/utils/test-fixture.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * This file overwrites the default playwright test fixture
+ * to apply some global settings and route handling.
+ */
+export const test = base.extend({
+  page: async ({ page }, use) => {
+    // Intercept /api/event requests
+    await page.route("/api/event", (route) => route.abort())
+    // Intercept  provider APIs
+    await page.route("**.jamendo.com**", (route) => route.abort())
+    await page.route("**.freesound.**", (route) => route.abort())
```
Comment (AetherUnbound) [inline frontend/test/playwright/utils/test-fixture.ts:14]: Just curious, why only these two?

**Review by AetherUnbound — CHANGES_REQUESTED** [review]
Hmm, the `eslint` rule doesn't appear to be working for me 😮 I made the following change:

```diff
diff --git a/frontend/test/storybook/visual-regression/v-select-field.spec.ts b/frontend/test/storybook/visual-regression/v-select-field.spec.ts
index c759c64af..b0265a671 100644
--- a/frontend/test/storybook/visual-regression/v-select-field.spec.ts
+++ b/frontend/test/storybook/visual-regression/v-select-field.spec.ts
@@ -1,6 +1,4 @@
-import { Page, expect } from "@playwright/test"
-
-import { test } from "~~/test/playwright/utils/test-fixture"
+import { Page, expect, test } from "@playwright/test"
 
 import { makeGotoWithArgs } from "~~/test/storybook/utils/args"
```

But `just lint eslint` is still passing. Is that expected?

File: frontend/test/playwright/utils/test-fixture.ts (hunk 3 above)
Comment (zackkrida) [reply frontend/test/playwright/utils/test-fixture.ts:14]: These were the only two that existed, my guess would be that the tests only use sample audio data from these providers but perhaps someone else on @WordPress/openverse-frontend can confirm.

**Comment by zackkrida** [comment]
@AetherUnbound I will look into the eslint rule! It _might_ be that you need to `just install` prior to testing, but I'll verify myself.

**Comment by sarayourfriend** [comment]
@zackkrida The ESLint rule you added is in the overrides block scoped exclusively to the unit tests (see the `files` setting). You'll need to move it to a different block that targets the playwright tests (`frontend/test/{playwright,storybook}/**/*.spec.{ts,js}` would do it, I think).

**Comment by zackkrida** [comment]
@sarayourfriend, ah, thanks so much! That's definitely it

File: frontend/test/playwright/utils/test-fixture.ts (hunk 3 above)
Comment (zackkrida) [reply frontend/test/playwright/utils/test-fixture.ts:14]: That's probably an excellent idea, I'll create a separate issue.

**Comment by zackkrida** [comment]
@AetherUnbound thanks to Sara's excellent suggestion this is ready for review 👍 the rule should definitely work now.

File: .eslintrc.js [hunk 4]
```diff
@@ -209,6 +210,27 @@ module.exports = {
             message:
               "Do not use @vue/test-utils' `shallowMount`. Use `~~/test/unit/test-utils/render` instead which includes helpful context setup or @testing-library/vue's `render` directly.",
           },
+          {
+            selector:
+              "ImportDeclaration[source.value='@playwright/test']:has(ImportSpecifier[local.name='test'])",
+            message:
+              "Do not import 'test' from '@playwright/test'. Use 'import { test } from '~~/test/playwright/utils/test-fixture' instead, which mocks Openverse provider APIs and the analytics server.",
+          },
```
[the remark this answers is not in the document]
Comment (zackkrida) [reply .eslintrc.js]: Yup, oops! Just forgot to delete.

**Review by sarayourfriend — APPROVED** [review]

**Comment by sarayourfriend** [comment]
LGTM once the tests pass :+1:

**Review by AetherUnbound — APPROVED** [review]
Confirmed the lint rule now works (read: breaks correctly) for me! 😄

**Comment by zackkrida** [comment]
Looks like there might be some test flake here 😢 I'll confirm on the next run and create the necessary critical issues.

**Pull request merged**